### PR TITLE
[12.0][IMP] stock_picking_invoicing set name from moves instead of product

### PR DESCRIPTION
**Pull request by sergiocorato**
Get name and uom from stock move to put in created invoice line.
In the same way, get price and discount from purchase line if exists and link invoice to it.

### CODE REVIEW

File: stock_picking_invoicing/wizards/stock_invoice_onshipping.py
Comment (kittiu) [inline stock_picking_invoicing/wizards/stock_invoice_onshipping.py]: As per naming convension, I think if this is not holding integer, it should be "uom", not "uom_id", is it not?

File: stock_picking_invoicing/wizards/stock_invoice_onshipping.py
Comment (kittiu) [inline stock_picking_invoicing/wizards/stock_invoice_onshipping.py]: Same as uom, should it be purchase_line, and not purchase_line_id ?

File: stock_picking_invoicing/wizards/stock_invoice_onshipping.py
Comment (kittiu) [inline stock_picking_invoicing/wizards/stock_invoice_onshipping.py:458]: Just curious, do we need to update values "name" again? _simulate_invoice_line_onchange() removed it?

**Review by kittiu — COMMENTED** [review]
Code review only

File: stock_picking_invoicing/wizards/stock_invoice_onshipping.py
Comment (sergiocorato) [reply stock_picking_invoicing/wizards/stock_invoice_onshipping.py:458]: Yes, [here](https://github.com/odoo/odoo/blob/15af5f47f1e3f25b222b6de1b084f8fee90935c2/addons/account/models/account_invoice.py#L1852)

File: stock_picking_invoicing/wizards/stock_invoice_onshipping.py
Comment (sergiocorato) [reply stock_picking_invoicing/wizards/stock_invoice_onshipping.py]: Done

File: stock_picking_invoicing/wizards/stock_invoice_onshipping.py
Comment (sergiocorato) [reply stock_picking_invoicing/wizards/stock_invoice_onshipping.py]: Done, thanks!

**Review by kittiu — APPROVED** [review]

**Comment by sergiocorato** [comment]
@kittiu I removed purchase part as module does not depend on it, I'll put it in a glue module see #722

**Comment by sergiocorato** [comment]
@kittiu merge?

**Comment by kittiu** [comment]
@sergiocorato we need one more approver to merge.

**Comment by sergiocorato** [comment]
@kittiu As of [contributing convention](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/oca_module_lifecycle_development_status.rst) one review is sufficient.

**Comment by kittiu** [comment]
> @kittiu As of [contributing convention](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/oca_module_lifecycle_development_status.rst) one review is sufficient.

OK, as highlighed in this statement for Beta - `Are recommended to have at least one peer review before merging PRs`

Thank you, I mostly forgot.

**Pull request merged**
